server/src/main/java/org/elasticsearch/common/xcontent/LoggingDeprecationHandler.java

```diff
@@ -49,17 +51,33 @@ private LoggingDeprecationHandler() {
     }
 
     @Override
-    public void usedDeprecatedName(String usedName, String modernName) {
-        deprecationLogger.deprecated("Deprecated field [{}] used, expected [{}] instead", usedName, modernName);
+    public void usedDeprecatedName(String parserName, Supplier<XContentLocation> location, String usedName, String modernName) {
+        if (parserName != null) {
+            deprecationLogger.deprecated("[{}][{}] Deprecated field [{}] used, expected [{}] instead",
+                parserName, location.get(), usedName, modernName);
+        } else {
+            deprecationLogger.deprecated("Deprecated field [{}] used, expected [{}] instead", usedName, modernName);
+        }
```

Comment:
The only suggestion I'd make is about whether it's possible to consolidate the log messages. For example:

```suggestion
        String prefix = parserName == null ? "" :
            "[" + parserName + "][" + location.get() + "] ";
        deprecationLogger.deprecated("{}Deprecated field [{}] used, expected [{}] instead", prefix, usedName, modernName);
```